sdk/src/clap_feature.rs

```diff
@@ -58,3 +58,37 @@ impl clap::builder::TypedValueParser for ByteCountParser {
         parse(cmd, arg, value).ok_or_else(|| clap::Error::new(clap::error::ErrorKind::InvalidValue))
     }
 }
+
+impl clap::builder::ValueParserFactory for types::BlockSize {
+    type Parser = BlockSizeParser;
+    fn value_parser() -> Self::Parser {
+        BlockSizeParser
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct BlockSizeParser;
+impl clap::builder::TypedValueParser for BlockSizeParser {
+    type Value = types::BlockSize;
+
+    fn parse_ref(
+        &self,
+        cmd: &clap::Command,
+        arg: Option<&clap::Arg>,
+        value: &std::ffi::OsStr,
+    ) -> Result<Self::Value, clap::Error> {
+        fn parse(
+            _cmd: &clap::Command,
+            _arg: Option<&clap::Arg>,
+            value: &std::ffi::OsStr,
+        ) -> Option<types::BlockSize> {
+            let value: Option<i64> = value.to_str()?.parse().ok();
+            match value {
+                Some(v) => v.try_into().ok(),
+                None => None,
+            }
+        }
+
+        parse(cmd, arg, value).ok_or_else(|| clap::Error::new(clap::error::ErrorKind::InvalidValue))
+    }
```

Comment:
why was this needed?

Reply:
ok; you might want to just do the validation inline: "512", "2048", "2k" ... (not sure why we support 2k, but I'm sure there are reasons...)

Reply:
> do the validation inline

I'm not 100% sure what you mean by this but my assumption is

```patch
diff --git a/sdk/src/clap_feature.rs b/sdk/src/clap_feature.rs
index 04dfc65..9348fa7 100644
--- a/sdk/src/clap_feature.rs
+++ b/sdk/src/clap_feature.rs
@@ -84,11 +84,16 @@ impl clap::builder::TypedValueParser for BlockSizeParser {
         ) -> Option<types::BlockSize> {
             let value: Option<i64> = value.to_str()?.parse().ok();
             match value {
-                Some(v) => v.try_into().ok(),
+                Some(512) => 512.try_into().ok(),
+                Some(2048) => 2048.try_into().ok(),
+                Some(4096) => 4096.try_into().ok(),
+                Some(_) => None,
                 None => None,
             }
         }
```

which I've committed (along with a better error message) in 6847f20